### PR TITLE
[api] correct signature's V value in returned web3 transaction

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
When web3 returns a tx, its signature's V value is not correctly set. Existing func in `action` package is used to generate the correct value

Fixes (#3944)

this PR will be needed for v1.12.1 to fix the API issue

## Type of change
Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/web3server_marshal_test.go [hunk 1]
```diff
@@ -251,7 +254,7 @@ func TestTransactionObjectMarshal(t *testing.T) {
 			"input":"0x",
 			"r":"0x3639643839613061663237646361613637663162363261333833353934643937",
 			"s":"0x3539396161646432623762313634636234313132616138646466643432663839",
-			"v":"0x35"
+			"v":"0x24c7"
```
Comment (dustinxie) [inline api/web3server_marshal_test.go:257]: this is the correct value corresponding to chainID = 4690

File: api/web3server_utils.go [hunk 2]
```diff
@@ -117,13 +117,16 @@ func (svr *web3Handler) getTransactionFromActionInfo(blkHash hash.Hash256, selp
...
+		return nil, err
+	}
 	return &getTransactionResult{
 		blockHash: blkHash,
 		to:        to,
-		ethTx:     ethTx,
+		ethTx:     tx,
 		receipt:   receipt,
 		pubkey:    selp.SrcPubkey(),
-		signature: selp.Signature(),
```
Comment (CoderZhi) [inline api/web3server_utils.go:126]: why could signature be deleted?

Comment (dustinxie) [reply api/web3server_utils.go:126]: not needed anymore, now the signed tx is generated, which contains correct signature value

File: api/web3server_marshal.go [hunk 3]
```diff
@@ -229,12 +227,7 @@ func (obj *getTransactionResult) MarshalJSON() ([]byte, error) {
 	}
 	value, _ := intStrToHex(obj.ethTx.Value().String())
 	gasPrice, _ := intStrToHex(obj.ethTx.GasPrice().String())
-
-	// TODO: if transaction is support EIP-155, we need to add chainID to v
-	vVal := uint64(obj.signature[64])
-	if vVal < 27 {
-		vVal += 27
-	}
+	v, r, s := obj.ethTx.RawSignatureValues()
```
Comment (envestcc) [inline api/web3server_marshal.go:230]: This change may not be compatible when chainID is not 0, please add a test to verify

Comment (dustinxie) [reply api/web3server_marshal.go:230]: it is compatible, and verified using the specific tx that you run into, added a test in latest commit

File: action/rlp_tx_test.go [hunk 4]
```diff
@@ -289,26 +288,8 @@ func TestRlpDecodeVerify(t *testing.T) {
...
-		sig := make([]byte, 64, 65)
-		rSize := len(r.Bytes())
-		copy(sig[32-rSize:32], r.Bytes())
-		sSize := len(s.Bytes())
-		copy(sig[64-sSize:], s.Bytes())
-		sig = append(sig, byte(recID))
-
-		// recover public key
-		rawHash := types.NewEIP155Signer(big.NewInt(int64(_evmNetworkID))).Hash(&tx)
-		pubkey, err := crypto.RecoverPubkey(rawHash[:], sig)
```
Comment (dustinxie) [inline action/rlp_tx_test.go:311]: duplicate code, same as `DecodeRawTx`

File: api/web3server_marshal.go [hunk 5]
```diff
@@ -229,12 +227,7 @@ func (obj *getTransactionResult) MarshalJSON() ([]byte, error) {
 	}
 	value, _ := intStrToHex(obj.ethTx.Value().String())
 	gasPrice, _ := intStrToHex(obj.ethTx.GasPrice().String())
-
-	// TODO: if transaction is support EIP-155, we need to add chainID to v
-	vVal := uint64(obj.signature[64])
-	if vVal < 27 {
-		vVal += 27
```
Comment (Liuhaai) [inline api/web3server_marshal.go:236]: IMO, its enough to add chainID with one line change here

File: api/web3server_utils.go [hunk 6]
```diff
@@ -117,13 +117,16 @@ func (svr *web3Handler) getTransactionFromActionInfo(blkHash hash.Hash256, selp
 	if err != nil {
 		return nil, err
 	}
+	tx, err := action.RawTxToSignedTx(ethTx, svr.coreService.ChainID(), selp.Signature())
```
Comment (Liuhaai) [inline api/web3server_utils.go:120]: tx.WithSignature(...) should be called inside ToEthTx(), which returns an unsigned tx incorrectly, instead?

Comment (dustinxie) [reply api/web3server_utils.go:120]: `ToEthTx()` is designed to return the raw/unsigned tx, which is passed to `rlpRawHash()` to compute the raw hash (to be signed), at this moment there's no signature yet

**Review by envestcc — APPROVED** [review]

**Pull request merged**